src/extension.ts

```diff
@@ -0,0 +1,151 @@
+import * as os from 'os';
+import { workspace, ExtensionContext, window, commands } from 'vscode';
+
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+  TransportKind
+} from 'vscode-languageclient/node';
+import { promisify } from 'util';
+import { execFile } from 'child_process';
+
+let client: LanguageClient;
+
+const outputChannel = window.createOutputChannel('Wasp Language Server');
+
+export async function activate(context: ExtensionContext) {
+  console.log('..Extension "vscode-wasp" is now active..');
+
+  const config = workspace.getConfiguration('wasp');
+
+  // Compute path to `waspls` executable, based on settings
+  const userDefinedExecutablePath = config.server.executable;
+  let executablePath = userDefinedExecutablePath === '' ? 'waspls' : userDefinedExecutablePath;
+
+  // Substitute vscode path variables into configuration
+  console.log(`Trying to find the server executable in ${executablePath}`);
+  executablePath = executablePath
+    .replace('${HOME}', os.homedir)
+    .replace('${home}', os.homedir)
+    .replace(/^~/, os.homedir);
+  if (executablePath === '') {
+    window.showErrorMessage(`wasp executable path ${executablePath} is empty, check your configuration`);
+    return;
+  }
+
+  let folders = workspace.workspaceFolders;
+  if (folders) {
+    let folder = folders[0];
+    if (folder) {
+      executablePath = executablePath.replace('${workspaceFolder}', folder.uri.path).replace('${workspaceRoot}', folder.uri.path);
+    }
+  }
```

Comment:
Why is this needed? Who is defining these vars?

Reply:
VSCode has some built-in variables like these that they substitute in in their configuration properties, but they don't provide any API for resolving the variables. Extensions commonly interpolate in these variables.

Reply:
Ok! Maybe leave a small comment.